storagenode/storagenodedb/infodb.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenodedb
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/migrate"
+)
+
+var ErrInfo = errs.Class("infodb")
+
+type infodb struct {
+	mu sync.Mutex
+	db *sql.DB
+}
+
+func NewInfo(path string) (*infodb, error) {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
+		return nil, err
+	}
+
+	db, err := sql.Open("sqlite3", "file:"+path+"?_journal=WAL")
+	if err != nil {
+		return nil, ErrInfo.Wrap(err)
+	}
+
+	return &infodb{db: db}, nil
+}
+
+func NewInfoInMemory() (*infodb, error) {
+	db, err := sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		return nil, ErrInfo.Wrap(err)
+	}
+
+	return &infodb{db: db}, nil
+}
+
+// Close closes any resources.
+func (db *infodb) Close() error {
+	return db.db.Close()
+}
+
+func (db *infodb) locked() func() {
+	db.mu.Lock()
+	return db.mu.Unlock
+}
+
+// CreateTables creates any necessary tables.
+func (db *infodb) CreateTables(log *zap.Logger) error {
+	migration := db.Migration()
+	return migration.Run(log.Named("migration"), db)
+}
+
+// RawDB returns access to the raw database, only for migration tests.
+func (db *infodb) RawDB() *sql.DB { return db.db }
+
+// Begin begins transaction
+func (db *infodb) Begin() (*sql.Tx, error) { return db.db.Begin() }
+
+// Rebind rebind parameters
+func (db *infodb) Rebind(s string) string { return s }
+
+// Schema returns schema
+func (db *infodb) Schema() string { return "" }
+
+// Migration returns table migrations.
+func (db *infodb) Migration() *migrate.Migration {
+	return &migrate.Migration{
+		Table: "versions",
+		Steps: []*migrate.Step{
+			{
+				Description: "Initial setup",
+				Version:     0,
+				Action: migrate.SQL{
+					// table for keeping serials that need to be verified against
+					`CREATE TABLE used_serial (
+						satellite_id  BLOB NOT NULL,
+						serial_number BLOB NOT NULL,
+						expiration    TIMESTAMP NOT NULL
+					)`,
+					// primary key on satellite id and serial number
+					`CREATE UNIQUE INDEX pk_used_serial ON used_serial(satellite_id, serial_number)`,
+					// expiration index to allow fast deletion
+					`CREATE INDEX idx_used_serial ON used_serial(expiration)`,
+
+					// certificate table for storing uplink/satellite certificates
+					`CREATE TABLE certificate (
+						cert_id       INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
```

Comment:
Maybe you explained earlier but why we need `cert_id`? Won't node_id will unique?

Reply:
Because a single node_id can have multiple leaf certificates.